src/data/set/basic.lean

```diff
@@ -560,6 +560,9 @@ by finish [subset_def, ext_iff, iff_def]
 theorem inter_eq_self_of_subset_right {s t : set α} (h : t ⊆ s) : s ∩ t = t :=
 by finish [subset_def, ext_iff, iff_def]
 
+theorem inter_eq_self_of_subset_iff {s t : set α} : s ⊆ t ↔ s ∩ t = s :=
```

Comment:
```suggestion
theorem subset_iff_inter_eq_self {s t : set α} : s ⊆ t ↔ s ∩ t = s :=
```
The lemma should reflect the shape of the statement: it's an iff statement, involving subset on the left and intersection on the right, hence the suggested name.